Sprint11 staging

rmathew1011 opened the pull request:
Resolves #485

rmathew1011 committed:
485 - to display the middle name.

 - Removed extra space from the beginning.
 - Used inline handlebar to conditionally render the irst name and the middle name.

rmathew1011 committed:
WIP - adding the handlebar helper for name display

rmathew1011 committed:
Added allNamesPresent handler bar to conditionally render name field

rmathew1011 committed:
Update mainContentTemplate.html

The changes to the span class styles were unintentional and only made for testing purposes. I’ve reverted them back to 
`<span class="h3"></span>`
since the style updates were not requested.

rmathew1011 committed:
WIP  - Dimensions id section is pending

rmathew1011 committed:
Edited the css styles for the researcher identifier section  - twitter

rmathew1011 committed:
Merge pull request #487 from TAMULib/researcher-identifier-section

Researcher identifier section

rmathew1011 committed:
WIP - added the dimensionId property for reseacher

rmathew1011 committed:
Removed the string ur from the dimensionsID

rmathew1011 committed:
WIP

rmathew1011 committed:
WIP - edited the styles in aside template file

rmathew1011 committed:
WIP

rmathew1011 committed:
Merge pull request #489 from TAMULib/researcher-identifier-dimensionsId

Researcher identifier dimensions

rmathew1011 committed:
Merge pull request #486 from TAMULib/485-display-middle-name

485 display middle name

tamu-sad-iii commented on src/main/java/edu/tamu/scholars/middleware/discovery/model/Person.java:
```diff
@@ -1135,7 +1142,7 @@ public class Person extends Common {
     @FieldSource(
         template = "person/middleName",
-        predicate = "http://www.w3.org/2006/vcard/ns#middleName"
+        predicate = "http://vivoweb.org/ontology/core#middleName"
```
Could we please take note to discuss with the product owner whether `firstName`, `middleName`, and/or `lastName` should be including in the default search field `_text_`?

See https://github.com/TAMULib/scholars-discovery/blob/cb1b7c6e067ec8e7fdc7f7f8011f9f2b73094ae0/src/main/java/edu/tamu/scholars/middleware/discovery/model/Person.java#L24

Currently, `name` is copied to the default search field and a `name_sort` fields to provide sorting against the whole string and not the tokenized field.

tamu-sad-iii commented on src/main/resources/defaults/displayViews/persons/asideTemplate.html:
```diff
@@ -82,30 +82,58 @@
+
+  {{#if twitter}}
+    <div class="id-row">
+      <a href="https://x.com/{{twitter}}" target="_blank" target="_blank" class="id-link">
```
Please remove the duplicate target attribute.

rmathew1011 replied:
Removed the duplicate target attribute.

tamu-sad-iii commented on src/main/resources/defaults/displayViews/persons/asideTemplate.html:
```diff
@@ -82,30 +82,58 @@
-      </span>
+  {{#if isiResearcherId}}
+  <div class="id-row">
+    <a href="https://www.webofscience.com/wos/author/record/{{isiResearcherId}}" target="_blank" class="id-link">
```
Are we using `http://vivo.library.tamu.edu/ontology/TAMU#ResearcherId`? If not, it would be ideal to refactor and only use `http://vivoweb.org/ontology/core#researcherId` as `researcherId` for the Web of Science identifier.

tamu-sad-iii reviewed (approved):
This is also resolving https://github.com/TAMULib/scholars-discovery/issues/488.

tamu-sad-iii commented on src/main/resources/defaults/displayViews/persons/asideTemplate.html:
```diff
@@ -82,30 +82,58 @@
+  {{#if orcidId}}
+  <div class="id-row">
+      <a href="https://orcid.org/{{orcidId}}" target="_blank" class="id-link">
+        <img src="/assets/images/tamu/orcid.svg" alt="orcidId" class="id-icon" />
```
@JamesS-TAMU noticed these should be relative path.

Please remove the leading forward slash for all img src hosted in scholars-angular assets.

tamu-sad-iii replied:
<img width="1919" height="1010" alt="image" src="https://github.com/user-attachments/assets/6ea95033-6032-4cd7-a263-410259219250" />

rmathew1011 replied:
leading slash removed for all images hosted in assets.

tamu-sad-iii reviewed (changes requested):
Please correct the research identifier icon image src to be relative path.

rmathew1011 committed:
This branch fixes PR suggestions and resolves mvn test failure:

Test Failure Fix:
PersonTest failed in edu.tamu.scholars.middleware.discovery.model.
Resolution: Added getter and setter methods for the dimensionsId property.

Duplicate Attribute Fix:
William identified a duplicate target="_blank" in:

<a href="https://x.com/{{twitter}}" target="_blank" class="id-link">
Resolution: Removed the extra attribute.

Icons Rendering Path Fix:
James noted icons were not rendering due to a leading slash in their relative path.
Resolution: removed the leading slash.

rmathew1011 committed:
Merge pull request #494 from TAMULib/sprint11-staging-PR-fix

This branch fixes PR suggestions and resolves mvn test failure:

rmathew1011 committed:
- Added conditional logic with the `or` helper to display the Researcher Identifier label.

rmathew1011 committed:
Merge pull request #495 from TAMULib/sprint11-researcher-identifier-display-fix

Added `or` helper to display the Researcher Identifier label.
